executor-scripts/linux/static

```diff
@@ -26,34 +26,36 @@ configure_addresses() {
 			PEER=""
 		fi
 
-		if [ -z "${MOCK}" -a "${1}" = "del" ]; then
-			# When having multiple addresses set from the same prefix they might/will(?) be configured
-			# as 'secondary' and implicitly removed when removing the non-secondary address. This
-			# leads ip complaining about not being able to remove the secondaries as they are already
-			# gone. So we ignore errors while deconfiguring addresses as they liked occur when removing
-			# a vanish address anyway.
-			${MOCK} ip "${addrfam}" addr "${1}" "${addr}" ${PEER} dev "${IFACE}" 2>/dev/null
-		else
-			${MOCK} ip "${addrfam}" addr "${1}" "${addr}" ${PEER} dev "${IFACE}"
-		fi
+		${MOCK} ip "${addrfam}" addr add "${addr}" ${PEER} dev "${IFACE}"
 	done
 }
 
 configure_gateways() {
 	for gw in ${IF_GATEWAYS}; do
 		addrfam=$(addr_family ${gw})
-		${MOCK} ip "${addrfam}" route "${1}" default via "${gw}" ${VRF_TABLE} ${METRIC} dev "${IFACE}"
+		${MOCK} ip "${addrfam}" route add default via "${gw}" ${VRF_TABLE} ${METRIC} dev "${IFACE}"
 	done
 }
 
+flush() {
+	cmd="addr"
+	arg="dev ${IFACE}"
+
+	if [ -n "$IF_VRF_TABLE" ]; then
+		cmd="route"
```

Comment:
The `static` executor isn't run for VRF interfaces unless there's an `address` configured and if we down the interface any routers are removed by the Kernel implicitly, so I think this special case isn't needed.

Reply:
It should also get run if there is:

```
iface vrf-red
    vrf-table 1
    gateway 192.168.0.1
```

Reply:
The default route will be removed as well then the VRF is removed as all member interfaces which could carry the conncted network containing the next-hop will be gone :-)

Reply:
Okay I deleted it for now.  If it turns out we need it, we can always bring it back later.